src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -3294,15 +3334,47 @@ private static bool TryGetNumberValue(string strNum, bool hex, bool real, char s
                     style = hex ? NumberStyles.AllowHexSpecifier : NumberStyles.AllowLeadingSign;
 
                     long longValue;
-                    if (suffix == 'l' || suffix == 'L')
+                    switch (suffix)
                     {
-                        if (long.TryParse(strNum, style, NumberFormatInfo.InvariantInfo, out longValue))
-                        {
-                            result = longValue * multiplier;
-                            return true;
-                        }
-                        result = null;
-                        return false;
+                        case NumberSuffixFlags.Long:
```

Comment:
What about "ul"?

Reply:
Presently, that one lives under the default case, inside the checks for the Unsigned flags.

Reply:
Sorry case is not HasFlag. Closed.